pypit/arcoadd.py

```diff
@@ -438,6 +445,58 @@ def scale_spectra(spectra, smask, sn2, iref=0, scale_method='auto', hand_scale=N
     return scales, omethod
 
 
+def bspline_cr(spectra, n_grow_mask=1, cr_nsig=5.):
+    """ Experimental and not so successful..
+
+    Parameters
+    ----------
+    spectra
+    n_grow_mask
+    cr_nsig
+
+    Returns
+    -------
+
+    """
+    # Unpack
+    fluxes, sigs, wave = unpack_spec(spectra, all_wave=True)
+
+    # Concatenate
+    all_f = fluxes.flatten()
+    all_s = sigs.flatten()
+    all_w = wave.flatten()
+
+    # Sort
+    srt = np.argsort(all_w)
+
+    # Bad pix
+    goodp = all_s[srt] > 0.
+
+    # Fit
+    mask, bspl = arutils.robust_polyfit(all_w[srt][goodp], all_f[srt][goodp], 3,
+                                        function='bspline', sigma=cr_nsig, everyn=2*spectra.nspec,
+                                        weights=1./np.sqrt(all_s[srt][goodp]), maxone=False)
+    # Plot?
+    debug = True
+    if debug:
+        from matplotlib import pyplot as plt
+        plt.clf()
+        ax = plt.gca()
+        ax.scatter(all_w[srt][goodp], all_f[srt][goodp], color='k')
+        #
+        x = np.linspace(np.min(all_w), np.max(all_w), 30000)
+        y = arutils.func_val(bspl, x, 'bspline')
+        ax.plot(x,y)
+        # Masked
+        ax.scatter(all_w[srt][goodp][mask==1], all_f[srt][goodp][mask==1], color='r')
+        # Range
+        stdf = np.std(all_f[srt][goodp])
+        ax.set_ylim(-2*stdf, 3*stdf)
+        plt.show()
+        debugger.set_trace()
+
+    debugger.set_trace()
```

Comment:
Should this debugger be removed?

Reply:
removed, although this method is not really so good..